mqtt-streaming/src/main/scala/akka/stream/alpakka/mqtt/streaming/scaladsl/MqttSession.scala

```diff
@@ -181,204 +148,246 @@ final class ActorMqttClientSession(settings: MqttSessionSettings)(implicit mat:
+                      Source.fromFuture(
+                        reply.future.map(command => cp.encode(ByteString.newBuilder, command.packetId).result())
+                      )
+                    case Command(cp: Disconnect.type, _, _) =>
+                      val reply = Promise[ClientConnector.ForwardDisconnect.type]
+                      clientConnector ! ClientConnector.DisconnectReceivedLocally(connectionId, reply)
+                      Source.fromFuture(reply.future.map(_ => cp.encode(ByteString.newBuilder).result()))
+                    case c: Command[A] => throw new IllegalStateException(c + " is not a client command")
                   }
-                case Command(cp: Subscribe, _, carry) =>
-                  val reply = Promise[Subscriber.ForwardSubscribe]
-                  clientConnector ! ClientConnector.SubscribeReceivedLocally(connectionId, cp, carry, reply)
-                  Source.fromFuture(
-                    reply.future.map(command => cp.encode(ByteString.newBuilder, command.packetId).result())
-                  )
-                case Command(cp: Unsubscribe, _, carry) =>
-                  val reply = Promise[Unsubscriber.ForwardUnsubscribe]
-                  clientConnector ! ClientConnector.UnsubscribeReceivedLocally(connectionId, cp, carry, reply)
-                  Source.fromFuture(
-                    reply.future.map(command => cp.encode(ByteString.newBuilder, command.packetId).result())
-                  )
-                case Command(cp: Disconnect.type, _, _) =>
-                  val reply = Promise[ClientConnector.ForwardDisconnect.type]
-                  clientConnector ! ClientConnector.DisconnectReceivedLocally(connectionId, reply)
-                  Source.fromFuture(reply.future.map(_ => cp.encode(ByteString.newBuilder).result()))
-                case c: Command[A] => throw new IllegalStateException(c + " is not a client command")
-              }
-            )
-            .recover {
-              case _: WatchedActorTerminatedException => ByteString.empty
-            }
-            .filter(_.nonEmpty)
-            .log("client-commandFlow", _.iterator.decodeControlPacket(settings.maxPacketSize)) // we decode here so we can see the generated packet id
-            .withAttributes(ActorAttributes.logLevels(onFailure = Logging.DebugLevel))
-        )
+                )
+                .recover {
+                  case _: WatchedActorTerminatedException => ByteString.empty
+                }
+                .filter(_.nonEmpty)
+                .log("client-commandFlow", _.iterator.decodeControlPacket(settings.maxPacketSize)) // we decode here so we can see the generated packet id
+                .withAttributes(ActorAttributes.logLevels(onFailure = Logging.DebugLevel))
+          }
+          .recover {
+            case e: TimeoutException =>
+              val f: CommandFlow[A] =
+                Flow.fromSinkAndSource(Sink.ignore, Source.empty[ByteString])
+              f
```

Comment:
This hack makes the shutdown test
https://github.com/akka/alpakka/blob/9a87c963dc83fae259a372b59918f70e1f57e979/mqtt-streaming/src/test/scala/docs/scaladsl/MqttSessionSpec.scala#L1256
behave as before. But the correct behaviour should be I failure when someone tries to use a session that is shut down already, right?
cc @huntc